forch/faucetizer.py

```diff
@@ -178,9 +195,17 @@ def _faucetize(self):
                                mac, device_behavior.segment)
                 continue
 
+            # update port vlan and acls
             port_cfg['native_vlan'] = vid
             if device_behavior.role:
-                port_cfg['acls_in'] = [f'role_{device_behavior.role}']
+                acl_name = f'role_{device_behavior.role}'
+                if self._has_acl(acl_name):
+                    port_cfg['acls_in'] = [acl_name]
+                else:
+                    port_cfg['acls_in'] = []
+                    LOGGER.error('No ACL defined for role %s', device_behavior.role)
+            else:
+                port_cfg['acls_in'] = []
```

Comment:
The two lines below this seem a bit sketchy -- can you re-write?  "setdefault" seems redundant with the code above, and then if tail_acl is NOT defined then it's not called at all, which means that either 1) it's not necessary to call at all, or 2) there's some weird corner-case when it should be called and isn't.  So, I think you can just take it out and directly call .append when appropriate.

Reply:
Right! The setdefault is useless as 'acls_in' is always set when processing the role. PTAL...